extensions/amp-viafoura/0.1/amp-viafoura.js

```diff
@@ -38,40 +37,67 @@ class AmpViafoura extends AMP.BaseElement {
     return isLayoutSizeDefined(layout);
   }
 
+  getDataAttrs() {
+    return Array.prototype.slice.call(this.element.attributes)
+      .map(attr => attr.nodeName)
+      .filter(attrName => attrName.match('data-.*'))
```

Comment:
imho the first map is a little pointless, I'd just do filter(attr.nodeName.match).map(...)